webhook_to_fedora_messaging/move.py

```diff
@@ -0,0 +1,148 @@
+
+def read_user():
+    with open(userscsv_fileloca) as file:
+        useriter = csv.reader(file)
+        next(useriter)
+        for item in useriter:
+            yield item[0]
+
+
+def read_repo():
+    with open(reposcsv_fileloca) as file:
+        repoiter = csv.reader(file)
+        next(repoiter)
+        for item in repoiter:
+            yield item[1], item[2], item[3], item[4], item[5]
+
+
+def read_pair():
+    pair, userdict = dict(), dict()
+    with open(userscsv_fileloca) as file:
+        useriter = csv.reader(file)
+        next(useriter)
+        for item in useriter:
+            userdict[item[1]] = item[0]
+    with open(pairscsv_fileloca) as file:
+        pairiter = csv.reader(file)
+        next(pairiter)
+        for item in pairiter:
+            if item[1] not in pair:
+                pair[item[1]] = [userdict[item[0]]]
+            else:
+                pair[item[1]].append(userdict[item[0]])
+    return pair
+
+
+async def import_user_to_database():
+    async for sess in get_session():
+        for item in read_user():
+            try:
+                if "github_org" not in item:
+                    make_user = User(name=item)
+                    sess.add(make_user)
+                    await sess.flush()
+                    await sess.commit()
+                    print(f"[{datetime.now().strftime(format)}] User '{item}' created.")
+                else:
+                    print(f"[{datetime.now().strftime(format)}] User '{item}' abandoned.")
+            except Exception as expt:
+                await sess.rollback()
+                print(f"[{datetime.now().strftime(format)}] User '{item}' failed.")
+                print(f"An error occurred: {expt}")
+        await sess.close()
+
+
+async def import_repo_to_database():
+    pairdata, done = read_pair(), set()
+    async for sess in get_session():
+        for item in read_repo():
+            name, desc, lang, team, active = item
+            if active == "f":
+                continue
+
+            try:
+                flag = 0
+                if team in pairdata:
+                    flag = 1
+                else:
+                    try:
+                        flag = 2
+                        qery = select(User).filter_by(name=team).options(selectinload("*"))
+                        rslt = await sess.execute(qery)
+                        user = rslt.scalar_one()
+                    except NoResultFound:
+                        print(f"[{datetime.now().strftime(format)}] Service '{team}/{name}' abandoned.")
+                        continue
+
+                # SERVICE IS TO BE CREATED ONLY IF FLAG IS 1 OR 2
+                # ORPHANS ARE TO BE ABANDONED
+                make_service = Service(
+                    name=f"{team}/{name}",
+                    uuid=uuid4().hex[0:8],
+                    type="github",
+                    desc=f"Migrated from GitHub2FedMsg. " + f"Written in {lang}. {desc}" if lang != "None" else f"{desc}",
```

Comment:
I'm not sure there's a lot of value to store the language that Github detected in this DB, or the original project description. Just "Migrated from GitHub2FedMsg" would make it easier to search in the DB later (it avoids the slow `LIKE`). I've looked at the github2fedmsg code and it just gets the description from github, stores it, and does nothing else with it. We can drop it.

By the way, the `desc` column is set to NOT NULL, so it's mandatory. I'll change that.

Reply:
Interesting. I remember otherwise. But maybe I misremember.